### ISSUE TITLE
Feature request: flush results for each 'page'

### ISSUE THREAD

**Issue by lury** [AnalogJ/hatchet#6]
Hi,

I've noticed that process of downloading messages is split into 500 messages chunks.
Could you add export flushes to csv, to have results for each 'page' of processed data without waiting for the whole process to finish?

Regards,
Lury

**Comment by AwesomeShare** [comment]
Yes, this is needed.  I have over 614 thousand emails in my Gmail account and this will take forever to process all my emails.  It would be great if it could flush the results after 500 messages have been processed.

**Comment by AnalogJ** [comment]
This is a great idea. I have almost 400k emails in my inbox, and I had to let it run overnight. 

I'll need to think about the architecture of the state file, so this might take a bit to implement.
